### ISSUE TITLE
New-HPOVUplinkSet InvalidOperation

### ISSUE THREAD

**Issue by jrkirkwood** [HewlettPackard/POSH-HPOneView#313]
Please fill in as much information as possible to help resolve your issue.

### Expected Behavior
Calling New-HPOVUplinkSet with bound InputObject to create new Uplink Set.  Expect the call to complete as all the required information is included.

### Actual Behavior
Getting an error around the -Networks parameter.

### Steps to reproduce
Use this command to recreate:
```PowerShell
Get-HPOVLogicalInterconnectGroup -Name "ESXi Logical Interconnect Group" | New-HPOVUplinkSet -Name "Production B" -Type Ethernet -Networks "prod vm1 B","prod vm2 B","prod vm3 B","prod vm4 B" -UplinkPorts "BAY2:X2" -EthMode "Auto"

Error output:
GetNetworkUris : A positional parameter cannot be found that accepts argument 'prod vm1 B'.
At C:\Program Files\WindowsPowerShell\Modules\HPOneView.400\HPOneView.400.psm1:69530 char:22
+                     $_NetworkUris = GetNetworkUris -_Networks $Networks -_ApplianceConnection $ ...
+                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [GetNetworkUris], ParameterBindingException
    + FullyQualifiedErrorId : PositionalParameterNotFound,GetNetworkUris

New-HPOVUplinkSet : [Send-HPOVRequest]: An internal error occurred while attempting the operation. If the error
persists, contact your authorized support representative and provide them with a support dump.
At line:1 char:55
+ $task = Get-HPOVLogicalInterconnectGroup -Name $lig | New-HPOVUplinkSet -Name "P ...
+                                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (Send-HPOVRequest:String) [New-HPOVUplinkSet], InvalidOperationExcepti
   on
    + FullyQualifiedErrorId : CRM_DEFAULT_INTERNAL_SERVER_EXCEPTION,New-HPOVUplinkSet
```

Traced the issue to a Get-HPOVNetwork call in function GetNetworkUris (line 69646).  I updated the call on line 69693 of the HPOneView.400.psm1 module to resolve the issue:
OLD:  

```PowerShell
$_net = Get-HPOVNetwork $_net -ApplianceConnection $_ApplianceConnection
```
NEW: 
```PowerShell
 $_net = Get-HPOVNetwork -Name $_net -ApplianceConnection $_ApplianceConnection
```

### Version Information
HPE OneView PowerShell Library Version (`Get-HPOVVersion` or `$PSLibraryVersion`): 4.0.1587.3400 
HPE OneView Appliance Version (`Get-HPOVVersion -ApplianceVer`):  
```PowerShell
PS C:\OneViewDCS400_POSH> Get-HPOVVersion -ApplianceVer

10.197.15.131                           LibraryVersion                          Path
-------------                           --------------                          ----
HPOneView.Appliance.NodeInfo            4.0.1587.3400                           C:\Program Files\WindowsPowerShell\M...
```

Output from `$PSVersionTable` on your Windows Host:

```PowerShell
Name                           Value
----                           -----
PSVersion                      4.0
WSManStackVersion              3.0
SerializationVersion           1.1.0.1
CLRVersion                     4.0.30319.42000
BuildVersion                   6.3.9600.18773
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0}
PSRemotingProtocolVersion      2.2
```

**Comment by ChrisLynchHPE** [comment]
Please update your script to use objects instead of resource names.  Your call should be:

```PowerShell
$Nets = Get-HPOVNetwork -Name "prod vm* B" -ErrorAction Stop
Get-HPOVLogicalInterconnectGroup -Name "ESXi Logical Interconnect Group" | New-HPOVUplinkSet -Name "Production B" -Type Ethernet -Networks $Nets -UplinkPorts "BAY2:X2" -EthMode "Auto"
```

Or if you prefer a 1-liner:

```PowerShell
Get-HPOVLogicalInterconnectGroup -Name "ESXi Logical Interconnect Group" | New-HPOVUplinkSet -Name "Production B" -Type Ethernet -Networks (Get-HPOVNetwork -Name "prod vm* B" -ErrorAction Stop) -UplinkPorts "BAY2:X2" -EthMode "Auto"
```

The library has gone through changes, and supporting resource names in Cmdlets like `New-HPOVUplinkSet` is difficult to support with multiple appliance connections.

That being said, I can certainly fix that error, but its priority will be low.

**Comment by ChrisLynchHPE** [comment]
This is fixed in [Release 4.0.1612.2800](https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v4.0.1612.2800) and in [PowerShell Gallery](https://www.powershellgallery.com/packages/HPOneView.400/4.0.1612.2800).

Please verify before this issue is closed.

**Comment by ChrisLynchHPE** [comment]
Closing due to no further response.  If this issue is still not fixed for you, we can reopen the issue.
